clients/src/main/java/org/apache/kafka/common/security/authenticator/DefaultLogin.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.security.authenticator;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.Subject;
+
+import org.apache.kafka.common.security.JaasUtils;
+import org.apache.kafka.common.security.auth.Login;
+import org.apache.kafka.common.security.authenticator.SaslClientAuthenticator.ClientCallbackHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Base login class that implements methods common to typical SASL mechanisms.
+ */
+public class DefaultLogin implements Login {
+    private static final Logger log = LoggerFactory.getLogger(DefaultLogin.class);
+
+    private String loginContextName;
+    private LoginContext loginContext;
+    private final CallbackHandler callbackHandler;
+
+    public DefaultLogin() {
+        this.callbackHandler = new ClientCallbackHandler();
```

Comment:
@junrao Yes, it was like that with Kerberos, and I imagine the class was reused to avoid code duplication. But actually I think it is better to use a different class for Login to make the logic clearer and more readable. I have added a different callback handler in DefaultLogin with just the callbacks for Login. There is some overlap with the client callback handler. Let me know what you think.